src/OpenCensus/Trace/Propagation/TraceContextFormat.cs

```diff
@@ -0,0 +1,136 @@
+﻿// <copyright file="TraceContextFormat.cs" company="OpenCensus Authors">
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of theLicense at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenCensus.Trace.Propagation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using OpenCensus.Utils;
+
+    /// <summary>
+    /// W3C trace context text wire protocol formatter. See https://github.com/w3c/distributed-tracing/.
+    /// </summary>
+    public class TraceContextFormat : TextFormatBase
+    {
+        /// <inheritdoc/>
+        public override ISet<string> Fields => new HashSet<string> { "tracestate", "traceparent" };
+
+        /// <inheritdoc/>
+        public override ISpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> getter)
+        {
+            // from https://github.com/w3c/distributed-tracing/blob/master/trace_context/HTTP_HEADER_FORMAT.md
+            // traceparent: 00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01
+
+            var traceparent = getter(carrier, "traceparent")?.FirstOrDefault();
+            var tracestateCollection = getter(carrier, "tracestate");
+
+            if (traceparent == null)
+            {
+                return null;
+            }
+
+            // TODO: validate version prefix
+            var traceId = TraceId.FromBytes(Arrays.StringToByteArray(traceparent, "00-".Length, "0af7651916cd43dd8448eb211c80319c".Length));
```

Comment:
please add constants for traceparent segments length

Reply:
I like this code more. Compiler will make it a constant for me. Any other benefits of constant

Reply:
will it calculate the size once or every time? Also, if you accidentally used a wrong string here - nobody would ever notice ;)

Reply:
hm, I was sure compiler will optimize it on compilation. 


But I was wrong. This is an output in Release:
		IL_003c: ldstr "0af7651916cd43dd8448eb211c80319c"
		IL_0041: call instance int32 [mscorlib]System.String::get_Length()


Issue is active on GitHub: https://github.com/dotnet/coreclr/issues/3633

Reply:
I'll make constants for now